sklearn/tests/test_multioutput.py

```diff
@@ -358,6 +390,46 @@ def test_multiclass_multioutput_estimator_predict_proba():
         assert_almost_equal(y_result[i], y_actual[i])
 
 
+def test_multiclass_multioutput_estimator_decision_function():
+    seed = 542
+
+    # make test deterministic
+    rng = np.random.RandomState(seed)
+
+    # random features
+    X = rng.normal(size=(5, 5))
+
+    # random labels
+    y1 = np.array(["b", "a", "a", "b", "a"]).reshape(5, 1)  # 2 classes
+    y2 = np.array(["d", "e", "f", "e", "d"]).reshape(5, 1)  # 3 classes
+
+    Y = np.concatenate([y1, y2], axis=1)
+
+    clf = MultiOutputClassifier(
+        LogisticRegression(solver="liblinear", random_state=seed)
+    )
+
+    clf.fit(X, Y)
+
+    y_result = clf.decision_function(X)
+
+    y_actual = [
+        np.array([1.181305, -0.71706655, -0.18780802, 0.62414541, -1.02976684]),
+        np.array(
+            [
+                [0.40278396, -0.96074739, -0.93055779],
+                [-0.9333856, 1.49696158, -1.54819887],
+                [-1.35860664, -1.24917287, 1.34488183],
+                [-0.8923613, 0.3491747, -1.4809791],
+                [1.418385, -0.75766164, -2.19373733],
+            ]
+        ),
+    ]
```

Comment:
I think I would rather avoid hard-coding such computed values in the tests (when they are not easy to compute by another mean than by running the scikit-learn code itself).

Instead I would just check the shape and the dtype of the results and the fact that thresholding at 0.0 fancy index `clf.classes_[0]` recovers the expected `y1` when for the binary target and taking `.argmax(axis=1)` to fancy index `clf.classes_[1]` recovers the expected `y2` for the multiclass target.

Reply:
If it's not the case (but I think it is at a quick glance), it means that the base logistic regression models underfit.  One could just increase the number of random features and decrease the regularization (higher `C`) to make them overfit instead.